src/peep/peep.c

```diff
@@ -8100,16 +8359,37 @@ int peep_pathfind_choose_direction(sint16 x, sint16 y, uint8 z, rct_peep *peep)
 			_peepPathFindQueueRideIndex = false;
 			_peepPathFindNumJunctions = maxNumJunctions;
 
-			uint16 score = sub_69A997(x, y, height, 0, 0xFFFF, test_edge);
+			/* Initialise _peepPathFindHistory.
+			 * The pathfinding will only use elements
+			 * 1..maxNumJunctions, so the starting point is
+			 * placed in element 0 */
+			_peepPathFindHistory[0].x = (uint8)(x >> 5);
+			_peepPathFindHistory[0].y = (uint8)(y >> 5);
+			_peepPathFindHistory[0].z = (uint8)z;
+			for (int junctionIndex = 1; junctionIndex < 16; junctionIndex++) {
```

Comment:
Just use `memset` instead of this loop. In fact i would memset the whole structure then assign `[0]` to the init value.